### PR TITLE
add CLI, preloading ability and upgrade to node v16

### PR DESCRIPTION
**Pull request by rcooke-warwick**
Add preloading and the CLI onto the worker so that we can avoid having to send an image to the worker over balena cloud proxy

part of https://github.com/orgs/balena-os/projects/6/views/1

Change-type: patch

### CODE REVIEW

**Comment by rcooke-warwick** [comment]
@vipulgupta2048 check out `Flowzone / e2e (generic-aarch64, qemu, 1941514)` check run - false positive reported

**Comment by rcooke-warwick** [comment]
marked as draft to prevent merge by accident. Unsure if this is the changes in this PR that caused the failure

**Comment by rcooke-warwick** [comment]
when trying this out with the qemu worker - many changes must be made 

- in leviathan, the worker container must be made privileged - this also means removing some other parts of the docker compose
- extra volumes must be made on the qemu worker docker compose in leviathan
- entry.sh must be modified - as the container ill be in privileged mode, commands such as `mknod` will fail as modes will already exist on the host
- I'm assuming that to use docker in docker we must be in privileged mode as all preload implementations we have within a container have privileged mode enabled

**Comment by rcooke-warwick** [comment]
this is again an argument for using @Bucknalla idea to just have the preload container be seperate - we won't have to modify any permissions with the qemu worker which could lead to unpredictable behaviour. 

at least we figured out how to make the node version bump work @vipulgupta2048 so it was worth trying ;/

File: Dockerfile.template [hunk 1]
```diff
@@ -46,13 +45,16 @@ RUN apk add --no-cache \
...
+  mdadm util-linux \
+  docker
   
 # hadolint ignore=DL3018
 RUN apk add --no-cache uhubctl --repository=http://dl-cdn.alpinelinux.org/alpine/edge/community || true
 
 RUN pip3 install usbsdmux
 
+VOLUME /var/lib/docker
+
```
Comment (klutchell) [inline Dockerfile.template:57]: Can probably skip defining this here and just make sure we define it in the compose file.

**Pull request closed without merging**